frontend/pages/policies/ManagePoliciesPage/components/CalendarEventsModal/CalendarEventsModal.tsx

```diff
@@ -396,7 +323,15 @@ const CalendarEventsModal = ({
       className={baseClass}
       width="large"
     >
-      {configured ? renderConfiguredModal() : renderPlaceholderModal()}
+      <>
+        {configured ? renderConfiguredModal() : renderPlaceholderModal()}
+        {showPreviewCalendarEvent && (
+          <CalendarEventPreviewModal
+            onCancel={togglePreviewCalendarEvent}
+            policy={selectedPolicyToPreview}
+          />
+        )}
```

Comment:
Moved the calendar preview to be a nested modal inside CalendarEventsModal so that the PoliciesPaginatedList can maintain its state (otherwise it loses the page its on whenever the preview is opened).

Reply:
We've generally been avoiding nested modals as a product rule. Would it be feasible to include the
lost state in `PaginatedList`'s `useImperativeHandle`, and have the parent grab that state to
persist before rendering the preview modal, then pass it back in when it goes back to the paginated
list view?

If not, maybe just cosmetically hide the evidence of the parent modal when rendering the child? It
currently looks like:

Reply:
<img width="668" alt="Screenshot 2025-02-18 at 11 30 04 AM" src="https://github.com/user-attachments/assets/ba0fecc6-6f47-4840-b8f4-fa14630a4955" />

Reply:
Handled this by hiding the overlay for the first modal when the second is opened.